middleware_timeout.go

```diff
@@ -22,21 +22,14 @@ func Timeout(timeout time.Duration) contractshttp.Middleware {
 		done := make(chan struct{})
 
 		go func() {
-			defer func() {
-				if r := recover(); r != nil {
-					LogFacade.Request(ctx.Request()).Error(r)
-					// TODO can be customized in https://github.com/goravel/goravel/issues/521
-					_ = ctx.Response().Status(http.StatusInternalServerError).String("Internal Server Error").Render()
-				}
-				close(done)
-			}()
-
+			defer HandleRecover(ctx, globalRecoverCallback)
 			ctx.Request().Next()
+			close(done)
```

Comment:
The CI error is caused by this line, basically, the process will be blocked when moving this line here, gin has this issue as well, please create another PR to fix this: Keep this line in defer.